src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1126,7 +1126,11 @@ private Symbol UnwrapAlias(Symbol symbol, out AliasSymbol alias, DiagnosticBag d
                 diagnostics, basesBeingResolved, qualifierOpt, node, plainName, node.Arity, options);
             NamedTypeSymbol resultType;
 
-            if (isUnboundTypeExpr)
+            if ((Flags & BinderFlags.LightweightTypeConstraintBinding) != 0)
```

Comment:
>(Flags & BinderFlags.LightweightTypeConstraintBinding) != 0 [](start = 16, length = 59)

Should this check be moved into the original `else` (after `if (isUnboundTypeExpr)`)?